pkg/virt-controller/watch/migration.go

```diff
@@ -1889,6 +1892,21 @@ func (c *MigrationController) alertIfHostModelIsUnschedulable(vmi *virtv1.Virtua
 	}
 }
 
+func (c *MigrationController) hasOwnerVM(vmi *virtv1.VirtualMachineInstance) bool {
+	controllerRef := v1.GetControllerOf(vmi)
+	if controllerRef == nil || controllerRef.Kind != virtv1.VirtualMachineGroupVersionKind.Kind {
+		return false
+	}
+
+	obj, exists, _ := c.vmStore.GetByKey(vmi.Namespace + "/" + controllerRef.Name)
+	if !exists {
+		return false
+	}
+
+	ownerVM := obj.(*virtv1.VirtualMachine)
+	return controllerRef.UID == ownerVM.UID
```

Comment:
This is how it is already implemented also on `VMIController` as well
the fact a VMI has a owner, doesn't mean it is indeed a real VM that kubevirt controls
it could be added manually for example, validating it is in the store cache shows it is a real controlled VM

Reply:
@oshoval can you add some comment here about IPAMClaim garbage collector corner case of vm re-creation with same name ?

Reply:
yes, but this is not the place of this function, i will add in the code that handle it